modelcluster/queryset.py

```diff
@@ -7,24 +7,24 @@
 
 # Constructor for test functions that determine whether an object passes some boolean condition
 def test_exact(model, attribute_name, value):
-    field = model._meta.get_field(attribute_name)
-    # convert value to the correct python type for this field
-    typed_value = field.to_python(value)
-    if isinstance(typed_value, Model):
-        if typed_value.pk is None:
+    if isinstance(value, Model):
+        if value.pk is None:
             # comparing against an unsaved model, so objects need to match by reference
-            return lambda obj: getattr(obj, attribute_name) is typed_value
+            return lambda obj: getattr(obj, attribute_name) is value
         else:
             # comparing against a saved model; objects need to match by type and ID.
             # Additionally, where model inheritance is involved, we need to treat it as a
             # positive match if one is a subclass of the other
             def _test(obj):
                 other_value = getattr(obj, attribute_name)
-                if not (isinstance(typed_value, other_value.__class__) or isinstance(other_value, typed_value.__class__)):
+                if not (isinstance(value, other_value.__class__) or isinstance(other_value, typed_value.__class__)):
```

Comment:
Is `typed_value` defined in this branch? I think the second clause should be `isinstance(other_value, value.__class__)`.

Reply:
Oops, well spotted! Looks like we could do with some better test coverage here, then...